### PR TITLE
Replace Maven with Gradle

### PR DESCRIPTION
**Pull request by jasoet-gdp**
1. Now use Gradle instead of Maven
2. Gradle configuration based on @RobWin's https://github.com/RobWin/asciidoc2ghPages
3. Use Gradle Wrapper with version 2.7
4. ~~Github Pages plugin is enabled, but not used yet (To be discussed)~~ Disabled for now
5. Readme not Updated Yet.

### CODE REVIEW

**Comment by jasoet-gdp** [comment]
@danieldietrich @RobWin please check this Pull Request before i merge it.

Thanks :smile:

File: .gitignore [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
+# Maven
+target
+
+# Netbeans Ignore
+nbproject
+dist
+private
 
-# -- Java
-hs_err_pid*
```
Comment (ruslansennov) [inline .gitignore]: <code>hs_err_pid*</code> should be keep (?)

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,74 @@
+buildscript {
+
+    ext{
+        asciiDoctorGradlePluginVersion="1.5.2"
+        asciiDoctorJPdfVersion="1.5.0-alpha.10"
+        asciiDoctorJVersion="1.5.3.1"
+        gradleGitVersion="1.3.2"
+        gradleWatchPluginVersion="0.1.5"
+        livereloadGradlePluginVersion="0.2.1"
```
Comment (ruslansennov) [inline build.gradle]: all versions I prefer keep in <code>gradle.properties</code>

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        classpath "org.asciidoctor:asciidoctor-gradle-plugin:${asciiDoctorGradlePluginVersion}"
+        classpath "org.asciidoctor:asciidoctorj-pdf:${asciiDoctorJPdfVersion}"
+        classpath "org.asciidoctor:asciidoctorj:${asciiDoctorJVersion}"
+        classpath "org.ajoberstar:gradle-git:${gradleGitVersion}"
+        classpath "com.bluepapa32:gradle-watch-plugin:${gradleWatchPluginVersion}"
+        classpath "org.kordamp.gradle:livereload-gradle-plugin:${livereloadGradlePluginVersion}"
+    }
+}
+description = 'Javaslang Documentation'
+version = '2.0.0-beta'
```
Comment (ruslansennov) [inline build.gradle]: also should be in <code>gradle.properties</code>

File: .gitignore (hunk 1 above)
Comment (jasoet-gdp) [inline .gitignore]: Oh my bad. We shouldn't keep `hs_err_pid*`.

File: build.gradle (hunk 2 above)
Comment (jasoet-gdp) [inline build.gradle]: Noted, will fix it.

File: build.gradle (hunk 3 above)
Comment (jasoet-gdp) [inline build.gradle]: Noted, will fix it.

**Comment by jasoet-gdp** [comment]
@ruslansennov updated. Please check

**Comment by jasoet-gdp** [comment]
Sorry, Accidentally Closed

**Comment by ruslansennov** [comment]
```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/ruslan/proj/javaslang-docs/build.gradle' line: 37

* What went wrong:
A problem occurred evaluating root project 'javaslang-docs'.
> Could not find property 'githubUser' on root project 'javaslang-docs'.
```

**Comment by jasoet-gdp** [comment]
@ruslansennov some properties required by githubPages according to https://github.com/RobWin/asciidoc2ghPages 

I think its better if we disabled the plugin for now since we not decide to use github pages or other site yet.

**Comment by ruslansennov** [comment]
I believe when we switch on the plugin, it should be checked to avoid error messages

**Comment by jasoet-gdp** [comment]
Yes, i will more careful next time.

**Comment by RobWin** [comment]
I'm fine with the changes. Thx

**Comment by danieldietrich** [comment]
Thanks, looks great!

Using github pages sounds good. How does it work? Can we trigger the build on travis-ci on a pull/commit and deploy to the github pages? Is the url then `javaslang-docs.github.io`? I will create a subdomain `doc.javaslang.com` or maybe `javaslang.com/docs` to refer to the github pages.

You can merge this pull request. Afterwards we will setup the build infrastructure in a separate issue.

Many thx!

**Comment by danieldietrich** [comment]
:+1:

**Pull request merged**